nextcloud28: init at 28.0.0, nextcloud27: 27.1.4 -> 27.1.5, nextcloud26: 26.0.9 -> 26.0.10

dotlambda opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

dotlambda committed:
nextcloudPackages: update

dotlambda committed:
nextcloud28: init at 28.0.0

Ma27 reviewed (changes requested):
The module must be updated as well.
Can take care of that latr today.

dotlambda commented:
Anything else that needs to be changed?

SuperSandro2000 reviewed (commented):
Can you do https://github.com/NixOS/nixpkgs/pull/274000#discussion_r1425911424 ?

SuperSandro2000 reviewed (changes requested):
I've deployed this to my instance and next to a new optional index I've hit the following which we should fix

```
This instance is missing some recommended PHP modules. For improved performance and better compatibility it is highly recommended to install them: bz2. For more details see the [documentation ↗](https://docs.nextcloud.com/server/28/go.php?to=admin-php-modules).
```

SuperSandro2000 commented:
Also I am hitting https://github.com/nextcloud/server/issues/42254

SuperSandro2000 commented:
Also I noticed that my desktop client started to sync external storages which crippled performance of my local lan and nextcloud. I couldn't find a bug report on this though.

dotlambda commented:
> I've deployed this to my instance and next to a new optional index I've hit the following which we should fix
> 
> ```
> This instance is missing some recommended PHP modules. For improved performance and better compatibility it is highly recommended to install them: bz2. For more details see the [documentation ↗](https://docs.nextcloud.com/server/28/go.php?to=admin-php-modules).
> ```

I added all recommended PHP packages.

SuperSandro2000 commented on nixos/modules/services/web-apps/nextcloud.nix:
```diff
@@ -28,6 +28,7 @@ let
   phpPackage = cfg.phpPackage.buildEnv {
     extensions = { enabled, all }:
       (with all; enabled
+        ++ [ bz2 intl sodium ] # recommended
```
Small note: I've checked the php-with-extensions derivation used by nextcloud and intl and sodium where already in there but I don't think it hurts listing them twice.

Ma27 committed:
nextcloud26: 26.0.9 -> 26.0.10

ChangeLog: https://github.com/nextcloud/server/releases/tag/v26.0.10

Ma27 committed:
nextcloud27: 27.1.4 -> 27.1.5

ChangeLog: https://github.com/nextcloud/server/releases/tag/v27.1.5

Ma27 commented:
Also pushed the updates of nc27/nc26 here.

Ma27 committed:
nixos/nextcloud: updates for nc28

* Always use PHP 8.2: at the time of writing, Nextcloud also suggests to
  use 8.2 rather than 8.3 in the manual for v28.

  One contributing factor is probably that all plugins need new releases
  to declare PHP 8.3 support.

* Fix upgradeWarning for installing v27 now that v28 is out.

* Drop upgrade warning for v24. This one is EOL for quite a while
  already, so right now everybody should've switched (or carefully
  studied the release notes in case they were upgrading from <23.05) and
  we can clean up the module a little bit.

  v25 was dropped not so long ago, so if it's still referenced (because
  somebody didn't declare `services.nextcloud.package` and has
  `system.stateVersion = "22.11";`) it's appropriate to still give a
  specialized error.

Ma27 committed:
nixos/doc: mention nc28

Ma27 committed:
nextcloud28Packages.bookmarks: 13.1.1 -> 13.1.2

Fixes

    Could not resolve symfonyAdapter! Class "symfonyAdapter" does not exist

when accessing the app.

Ma27 reviewed (approved):
Just pushed a few minor changes:
* Updates to docs & release-notes
* update bookmarks app to latest patch-release since the former one was actually broken with nc28
* minor cleanups to the nextcloud module (see the commit message for a detailed explanation)

It seems to work fine for me so far.

Ma27 commented:
Does @RaitoBezarius have some capacity to review this?
I'd probably wait until tomorrow (unless someone requests more time to review of course) and merge then.

RaitoBezarius commented:
> Does @RaitoBezarius have some capacity to review this? I'd probably wait until tomorrow (unless someone requests more time to review of course) and merge then.

Testing now.

RaitoBezarius reviewed (approved):
Tested the new NC28 on a small-size production deployment (100G of data) on stable NixOS 23.11:

- Onlyoffice.
- Adding the classical missing indexes.
- The new Vue.js Files application.

Everything works, good to go for me.
